mijia/examples/readings.rs

```diff
@@ -0,0 +1,78 @@
+//! Example of how to subscribe to readings from one or more sensors.
+
+use eyre::Report;
+use mijia::{MijiaEvent, MijiaSession, SensorProps};
+use std::process::exit;
+use std::time::Duration;
+use tokio::stream::StreamExt;
+use tokio::time;
+
+const SCAN_DURATION: Duration = Duration::from_secs(5);
+
+#[tokio::main]
+async fn main() -> Result<(), Report> {
+    pretty_env_logger::init();
+
+    // If at least one command-line argument is given, we will only try to connect to sensors whose
+    // MAC address containts one of them as a sub-string.
+    let mut args = std::env::args();
+    let binary_name = args
+        .next()
+        .ok_or_else(|| eyre::eyre!("Binary name missing"))?;
+    let filters: Vec<_> = args.collect();
+
+    if filters
+        .iter()
+        .any(|f| f.contains(|c: char| !(c.is_ascii_hexdigit() || c == ':')))
+    {
+        eprintln!("Invalid MAC addresses {:?}", filters);
+        eprintln!("Usage:");
+        eprintln!("  {} [MAC address]...", binary_name);
+        exit(1);
+    }
```

Comment:
Maybe split this out into a function since it's not that interesting to the example (I feel like there is a copy-pasted version of this in another example. That could also be split out into a function, but it's fine for them to both be copy-paste)

Reply:
Done.